opentrons-ai-client/src/molecules/InputPrompt/index.tsx

```diff
@@ -162,93 +122,174 @@ export function InputPrompt(): JSX.Element {
     isUpdateOrCreateRequest: boolean = false,
     isRegenerateRequest: boolean = false
   ): Promise<void> => {
-    const newRequestId = uuidv4() + getPreFixText(isUpdateOrCreateRequest)
+    const newRequestId = `${uuidv4()}${getPreFixText(
+      isUpdateOrCreateRequest,
+      isNewProtocol
+    )}`
     setRequestId(newRequestId)
     const currentProtocolFormat = detectProtocolFormat(
       watchUserPrompt,
       chatHistory
     )
+
+    // Prepare files for upload - use multipart for efficiency
+    let validatedFiles: File[] = []
+    if (attachedFiles.length > 0 && !isUpdateOrCreateRequest) {
+      const files = prepareFilesForUpload()
+      if (files === null) {
+        return // Validation failed, error already shown to user
+      }
+      validatedFiles = files
+    }
+
     const userInput: ChatData = {
       requestId: newRequestId,
       role: 'user',
       reply: watchUserPrompt,
       protocol_format: currentProtocolFormat,
       attachments:
-        attachedFiles.length > 0
-          ? attachedFiles.map(file => ({
-              name: file.name,
-              type: getFileType(file),
-              content: '', // Content will be read separately if needed
-              size: file.size,
-            }))
+        validatedFiles.length > 0
+          ? validatedFiles.map(file => {
+              const fileType = getFileType(file)
+              if (!fileType) {
```

Comment:
same as the above since fileType returns a specific fileType or null.
so using null is type-safe.

```suggestion
              if (fileType === null) {
```

Reply:
done